### PR TITLE
0.11.0-rc0

### PR DESCRIPTION
**Pull request by swcurran**
Signed-off-by: Stephen Curran <swcurran@gmail.com>

### CODE REVIEW

**Comment by swcurran** [comment]
I have now -- Doh!! Have to do that to generate the docs, but have a script that SHOULD delete it after. Will verify.

**Comment by dbluhm** [comment]
Gotcha! I recommend doing a history rewrite, as a side note. Commits to delete previously committed files means that that copied folder still exists in the history, unfortunately, contributing to bloat of the git history.

**Comment by swcurran** [comment]
Unfortunately, such git-voodoo is beyond me.  If you know of a way to do it, please let me know. Happy to close this PR and start again, if that would help.

I have no tie to my fork (no pending PRs), so happy to delete it, create a new one and apply the changes from this PR.  Is that the easiest approach?

**Comment by dbluhm** [comment]
A bit wordy but here's a walkthrough from ChatGPT:

Alright, to achieve your objective, we'll make use of an interactive rebase. Please make sure you backup your repository or double-check your operations, as history rewriting can potentially cause data loss if done incorrectly. Also, avoid rewriting history on branches shared with others, as this can cause confusion and conflicts.

Here are the steps:

1. **Begin an interactive rebase:**  
You'll want to start the rebase from the commit *before* commit 1. If `commit1` has the hash `abc123` (replace this with the actual hash), you would type:
    ```bash
    git rebase -i abc123^
    ```

2. **Modify the commit list:**
   You should now see a text editor pop up with a list of commits looking something like this:
    ```
    pick abc123 commit 1 message
    pick def456 commit 2 message
    pick ghi789 commit 3 message
    ```
   To remove the folder from the first commit, edit the word `pick` to `edit` for the line with `commit 1`:
    ```
    edit abc123 commit 1 message
    pick def456 commit 2 message
    ```

   Save and exit the text editor.

3. **Remove the folder from commit 1:**  
Now you'll be at the state of `commit 1`. Remove the folder and amend the commit:
    ```bash
    git rm -r foldername   # replace foldername with the name of the folder
    git commit --amend
    ```

    If you want, you can also edit the commit message to better reflect the changes. Once you're done, save and exit.

4. **Continue the rebase:**  
To move on:
    ```bash
    git rebase --continue
    ```

5. **Remove commit 3:**  
Since commit 3 was just to undo the changes from commit 1, and we've already fixed commit 1, you don't need commit 3 anymore. Start another interactive rebase:
    ```bash
    git rebase -i abc123^
    ```
    And you'll again see a list of commits. Simply delete the line corresponding to `commit 3` to remove it from history:
    ```
    pick abc123 commit 1 amended message (or whatever it is now)
    pick def456 commit 2 message
    ```

   Save and exit the text editor.

   After this process, the rebase should be complete, and the folder will never have appeared in the repository's history.

6. **Push to remote (with caution):**  
If you've already pushed the old commits to a remote, you'll need to forcefully push these changes since you've rewritten history. 
    ```bash
    git push origin branchname --force
    ```

    ⚠️ **Warning:** Force pushing can overwrite changes in the remote that you don't have locally. Make sure you coordinate with any collaborators and ensure they reset their local branches to match the updated remote branch.

And that's it! The folder will no longer be present in the repository's history.

Small amendment from me (Daniel, not ChatGPT); on the first interactive rebase command, you can just put `drop` for the third commit and it will do the same thing as step 5

**Comment by swcurran** [comment]
Thanks!  However, it was much easier just to save the 4 edited files, delete the fork, create a new fork, and a create a new PR.  I think I got it right the second time.

I will save this in my notebook — hopefully I’ll never need it…
